cupyx/scipy/interpolate/_interpolate.py

```diff
@@ -0,0 +1,902 @@
+            vb = eval_poly_1<T>(
+                next_breakpoint - breakpoint, coef, interval,
+                idx, -1, c_dims, stride_0, stride_1);
+        }
+
+        if(interval == start_interval) {
+            va = eval_poly_1<T>(
+                a - breakpoint, coef, interval, idx, -1, c_dims,
+                stride_0, stride_1);
+        } else {
+            va = eval_poly_1<T>(
+                0, coef, interval, idx, -1, c_dims,
+                stride_0, stride_1);
+        }
+
+        vtot += (vb - va);
+    }
+
+    if(!asc) {
+        vtot = -vtot;
+    }
+
+    out[idx] = vtot;
+
+}
+"""
+
+PPOLY_MODULE = cupy.RawModule(
+    code=PPOLY_KERNEL, options=('-std=c++11',),
+    name_expressions=(
+        [f'eval_ppoly<{type_name}>' for type_name in TYPES] +
+        [f'fix_continuity<{type_name}>' for type_name in TYPES] +
+        [f'integrate<{type_name}>' for type_name in TYPES]))
+
+
+def _get_module_func(module, func_name, *template_args):
+    def _get_typename(dtype):
+        typename = get_typename(dtype)
+        if dtype.kind == 'c':
+            typename = 'thrust::' + typename
+        return typename
+    args_dtypes = [_get_typename(arg.dtype) for arg in template_args]
+    template = ', '.join(args_dtypes)
+    kernel_name = f'{func_name}<{template}>' if template_args else func_name
+    kernel = module.get_function(kernel_name)
+    return kernel
+
+
+def _ppoly_evaluate(c, x, xp, dx, extrapolate, out):
+    """
+    Evaluate a piecewise polynomial.
+
+    Parameters
+    ----------
+    c : ndarray, shape (k, m, n)
+        Coefficients local polynomials of order `k-1` in `m` intervals.
+        There are `n` polynomials in each interval.
+        Coefficient of highest order-term comes first.
+    x : ndarray, shape (m+1,)
+        Breakpoints of polynomials.
+    xp : ndarray, shape (r,)
+        Points to evaluate the piecewise polynomial at.
+    dx : int
+        Order of derivative to evaluate.  The derivative is evaluated
+        piecewise and may have discontinuities.
+    extrapolate : bint
+        Whether to extrapolate to out-of-bounds points based on first
+        and last intervals, or to return NaNs.
+    out : ndarray, shape (r, n)
+        Value of each polynomial at each of the input points.
+        This argument is modified in-place.
+    """
+    # Determine if the breakpoints are in ascending order or descending one
+    ascending = (x[x.shape[0] - 1] >= x[0]).item()
+
+    intervals = cupy.empty(xp.shape, dtype=cupy.int64)
+    interval_kernel = INTERVAL_MODULE.get_function('find_breakpoint_position')
+    interval_kernel(((xp.shape[0] + 128 - 1) // 128,), (128,),
+                    (x, xp, intervals, extrapolate, xp.shape[0], x.shape[0],
+                     ascending))
+
+    # Compute coefficient displacement stride (in elements)
+    c_shape = cupy.asarray(c.shape, dtype=cupy.int_)
+    c_strides = cupy.asarray(c.strides, dtype=cupy.int_) // c.itemsize
```

Comment:
`int_` types caused errors in https://github.com/cupy/cupy/pull/7195#issuecomment-1343939311, maybe better to pre-emptively use int64 , as suggested above?